<html>
<head>
  <meta name="navigation" content="Comparative Genomics">
  <title>Pairwise Alignments</title>
  <link rel="stylesheet" type="text/css" media="all" href="85_spreadsheet_table.css" />
  <link rel="stylesheet" type="text/css" media="all" href="95_static_content.css" />
</head>

<body>

<h2>Hordeum vulgare vs Triticum urartu LastZ alignments</h2>

<p>Hordeum vulgare (<i>Hordeum vulgare</i>, 030312v2) and Triticum urartu (<i>Triticum urartu</i>, GCA_000347455.1) were aligned using the LastZ alignment algorithm (<a href="http://www.bx.psu.edu/miller_lab/dist/README.lastz-1.02.00/README.lastz-1.02.00a.html">LastZ</a>) in Ensembl release 76. Hordeum vulgare  was used as the reference species. After running LastZ, the raw LastZ  alignment blocks are chained according to their location in both genomes. During the final netting process, the best sub-chain is chosen in each region on the reference species.</p>

<a href="/info/genome/compara/analyses.html" class="cp-external">Full list of 
pairwise alignments</a>


<h3 class="boxed">Configuration parameters</h3>

<table class="ss tint species-stats">
<tr class="bg2">
<td class="data" style="width:75%">Gap open penalty (O)</td>
<td class="value">400</td>
</tr>
<tr>
<td class="data">Gap extend penalty (E)</td>
<td class="value">30</td>
</tr>
<tr class="bg2">
<td class="data">HSP threshold (K)</td>
<td class="value">3000</td>
</tr>
<tr>
<td class="data">Threshold for gapped extension (L)</td>
<td class="value">3000</td>
</tr>
<tr class="bg2">
<td class="data">Threshold for alignments between gapped alignment blocks (H)</td>
<td class="value">2200</td>
</tr>
<tr>
<td class="data">Masking count (M)</td>
<td class="value"></td>
</tr>
<tr class="bg2">
<td class="data">Seed and Transition value (T)</td>
<td class="value">1</td>
</tr>
<tr class="bg2">
<td class="data">Additional parameters</td>
<td class="value">--ambiguous=iupac--identity=75..100--matchcount=1000</td>
</tr>
<tr>
<td class="data">Scoring matrix (Q)</td>
<td class="value">Default</td>
</tr>
</table>

<h3 class="boxed">Chunking parameters</h3>
<table style="text-align: left; width: 100%;" border="0" cellpadding="2"
cellspacing="2">
<tbody>
<tr>
<td style="vertical-align: top;">
</td>
<th style="vertical-align: top;">Hordeum vulgare
</th>
<th style="vertical-align: top;">Triticum urartu
</th>
</tr>
<tr>
<th style="vertical-align: middle;">Chunk size
</th>
<td style="vertical-align: top;">50,000,000
</td>
<td style="vertical-align: top;">100,100,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Overlap
</th>
<td style="vertical-align: top;">0
</td>
<td style="vertical-align: top;">100,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Group set size
</th>
<td style="vertical-align: top;">0
</td>
<td style="vertical-align: top;">801,000,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Masking options
</th>
<td style="vertical-align: top;">{default_soft_masking => 1}
</td>
<td style="vertical-align: top;">{default_soft_masking => 1}
</td>
</tr>
</tbody>
</table>


<h3>Results</h3>

<p>Number of alignment blocks: 287740</p>

<table style="text-align: left; width: 100%;" border="0" cellpadding="2"
cellspacing="2">
<tbody>
<tr>
<td style="vertical-align: top;">
</td>
<th style="vertical-align: top;">Genome coverage(bp)
</th>
<th style="vertical-align: top;">Coding exon coverage (bp)
</th>
</tr>
<tr>
<th style="vertical-align: middle;">Hordeum vulgare</th>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./images/pie_ref_align_9210.png"/>
</td>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./images/pie_ref_cod_align_9210.png"/>
</td>
</tr>

<tr>
<td></td>
<td style="vertical-align: top;">Covered: 208,911,653 out of 4,706,173,842
</td>
<td style="vertical-align: top;">Matches: 19,980,450 out of 27,615,842
</td>
</tr>

<tr>
<td></td>
<td style="vertical-align: top;">Uncovered: 4,497,262,189 out of 4,706,173,842
</td>
<td style="vertical-align: top;">Mis-matches: 1,419,528 out of 27,615,842
</td>
</tr>

<tr>
<td></td>
<td></td>
<td style="vertical-align: top;">Insertions: 182,243 out of 27,615,842
</td>
</tr>

<tr>
<td></td>
<td></td>
<td style="vertical-align: top;">Uncovered: 6,033,621 out of 27,615,842
</td>
</tr>

<tr>
<th style="vertical-align: middle;">Triticum urartu
</th>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./images/pie_non_ref_align_9210.png"/>
</td>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./images/pie_non_ref_cod_align_9210.png"/>
</td>
</tr>
<tr>
<td></td>
<td style="vertical-align: top;">Covered: 136,653,047 out of 3,747,163,292
</td>
<td style="vertical-align: top;">Matches: 21,631,646 out of 36,143,424
</td>
</tr>

<tr>
<td></td>
<td style="vertical-align: top;">Uncovered: 3,610,510,245 out of 3,747,163,292
</td>
<td style="vertical-align: top;">Mis-matches: 1,463,568 out of 36,143,424
</td>
</tr>

<tr>
<td></td>
<td></td>
<td style="vertical-align: top;">Insertions: 172,025 out of 36,143,424
</td>
</tr>

<tr>
<td></td>
<td></td>
<td style="vertical-align: top;">Uncovered: 12,876,185 out of 36,143,424
</td>
</tr>

</tbody>
</table>


</body>
</html>
