<html>
<head>
  <meta name="navigation" content="Comparative Genomics">
  <title>Pairwise Alignments</title>
  <link rel="stylesheet" type="text/css" media="all" href="85_spreadsheet_table.css" />
  <link rel="stylesheet" type="text/css" media="all" href="95_static_content.css" />
</head>

<body>

<h2>Zea mays vs Sorghum bicolor LastZ alignments</h2>

<p>Zea mays (<i>Zea mays</i>, AGPv3) and Sorghum bicolor (<i>Sorghum bicolor</i>, Sorbi1) were aligned using the LastZ alignment algorithm (<a href="http://www.bx.psu.edu/miller_lab/dist/README.lastz-1.02.00/README.lastz-1.02.00a.html">LastZ</a>) in Ensembl release 76. Zea mays  was used as the reference species. After running LastZ, the raw LastZ  alignment blocks are chained according to their location in both genomes. During the final netting process, the best sub-chain is chosen in each region on the reference species.</p>

<a href="/info/genome/compara/analyses.html" class="cp-external">Full list of 
pairwise alignments</a>


<h3 class="boxed">Configuration parameters</h3>

<table class="ss tint species-stats">
<tr class="bg2">
<td class="data" style="width:75%">Gap open penalty (O)</td>
<td class="value">400</td>
</tr>
<tr>
<td class="data">Gap extend penalty (E)</td>
<td class="value">30</td>
</tr>
<tr class="bg2">
<td class="data">HSP threshold (K)</td>
<td class="value">3000</td>
</tr>
<tr>
<td class="data">Threshold for gapped extension (L)</td>
<td class="value">3000</td>
</tr>
<tr class="bg2">
<td class="data">Threshold for alignments between gapped alignment blocks (H)</td>
<td class="value">2200</td>
</tr>
<tr>
<td class="data">Masking count (M)</td>
<td class="value"></td>
</tr>
<tr class="bg2">
<td class="data">Seed and Transition value (T)</td>
<td class="value">1</td>
</tr>
<tr class="bg2">
<td class="data">Additional parameters</td>
<td class="value">--ambiguous=iupac</td>
</tr>
<tr>
<td class="data">Scoring matrix (Q)</td>
<td class="value">Default</td>
</tr>
</table>

<h3 class="boxed">Chunking parameters</h3>
<table style="text-align: left; width: 100%;" border="0" cellpadding="2"
cellspacing="2">
<tbody>
<tr>
<td style="vertical-align: top;">
</td>
<th style="vertical-align: top;">Zea mays
</th>
<th style="vertical-align: top;">Sorghum bicolor
</th>
</tr>
<tr>
<th style="vertical-align: middle;">Chunk size
</th>
<td style="vertical-align: top;">50,000,000
</td>
<td style="vertical-align: top;">100,100,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Overlap
</th>
<td style="vertical-align: top;">0
</td>
<td style="vertical-align: top;">100,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Group set size
</th>
<td style="vertical-align: top;">0
</td>
<td style="vertical-align: top;">100,100,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Masking options
</th>
<td style="vertical-align: top;">{default_soft_masking => 1}
</td>
<td style="vertical-align: top;">{default_soft_masking => 1}
</td>
</tr>
</tbody>
</table>


<h3>Results</h3>

<p>Number of alignment blocks: 464518</p>

<table style="text-align: left; width: 100%;" border="0" cellpadding="2"
cellspacing="2">
<tbody>
<tr>
<td style="vertical-align: top;">
</td>
<th style="vertical-align: top;">Genome coverage(bp)
</th>
<th style="vertical-align: top;">Coding exon coverage (bp)
</th>
</tr>
<tr>
<th style="vertical-align: middle;">Zea mays</th>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./images/pie_ref_align_9130.png"/>
</td>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./images/pie_ref_cod_align_9130.png"/>
</td>
</tr>

<tr>
<td></td>
<td style="vertical-align: top;">Covered: 197,912,455 out of 2,067,864,162
</td>
<td style="vertical-align: top;">Matches: 36,365,767 out of 44,177,667
</td>
</tr>

<tr>
<td></td>
<td style="vertical-align: top;">Uncovered: 1,869,951,707 out of 2,067,864,162
</td>
<td style="vertical-align: top;">Mis-matches: 3,514,913 out of 44,177,667
</td>
</tr>

<tr>
<td></td>
<td></td>
<td style="vertical-align: top;">Insertions: 564,985 out of 44,177,667
</td>
</tr>

<tr>
<td></td>
<td></td>
<td style="vertical-align: top;">Uncovered: 3,732,002 out of 44,177,667
</td>
</tr>

<tr>
<th style="vertical-align: middle;">Sorghum bicolor
</th>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./images/pie_non_ref_align_9130.png"/>
</td>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./images/pie_non_ref_cod_align_9130.png"/>
</td>
</tr>
<tr>
<td></td>
<td style="vertical-align: top;">Covered: 104,223,553 out of 738,540,932
</td>
<td style="vertical-align: top;">Matches: 25,188,317 out of 40,003,363
</td>
</tr>

<tr>
<td></td>
<td style="vertical-align: top;">Uncovered: 634,317,379 out of 738,540,932
</td>
<td style="vertical-align: top;">Mis-matches: 2,224,542 out of 40,003,363
</td>
</tr>

<tr>
<td></td>
<td></td>
<td style="vertical-align: top;">Insertions: 489,856 out of 40,003,363
</td>
</tr>

<tr>
<td></td>
<td></td>
<td style="vertical-align: top;">Uncovered: 12,100,648 out of 40,003,363
</td>
</tr>

</tbody>
</table>


</body>
</html>
